.github/workflows/protect-file.yml

```diff
@@ -0,0 +1,37 @@
+name: Protect certain files to avoid further changes
+
+on:
+  push:
+    branches:
+      - 'master'
+      - '202[2-9][0-9][0-9]'
+  pull_request:
+    branches:
+      - 'master'
+      - '202[2-9][0-9][0-9]'
+
+jobs:
+  check-file:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Fetch base branch
+        run: |
+          git fetch origin ${GITHUB_BASE_REF}
+
+      - name: Check protected files
+        run: |
+          echo "Starting file check step"
+          protected_files=('src/sonic-config-engine/minigraph.py' 'files/build_templates/init_cfg.json.j2')
```

Comment:
This is not comprehensive source code files. minigraph.py has dependencies on other python files.